batch/data-plane/src/test/java/com/microsoft/azure/batch/BatchIntegrationTestBase.java

```diff
@@ -0,0 +1,517 @@
+        return builder.build();
+    }
+
+    protected RestClient buildPlaybackRestClient(ServiceClientCredentials credentials, String baseUrl) throws IOException {
+        return buildRestClient(new RestClient.Builder().withBaseUrl(baseUrl)
+                .withSerializerAdapter(new AzureJacksonAdapter())
+                .withResponseBuilderFactory(new ResponseBuilder.Factory() {
+                    private final AzureResponseBuilder.Factory baseFactory = new AzureResponseBuilder.Factory();
+                    @Override
+                    public <T, E extends RestException> ResponseBuilder<T, E> newInstance(SerializerAdapter<?> serializerAdapter) {
+                        return baseFactory.<T, E>newInstance(serializerAdapter).withThrowOnGet404(true);
+                    }
+                })
+                .withCredentials(credentials)
+                .withLogLevel(LogLevel.NONE)
+                .withNetworkInterceptor(new LoggingInterceptor(LogLevel.BODY_AND_HEADERS))
+                .withNetworkInterceptor(interceptorManager.initInterceptor())
+                .withInterceptor(new ResourceManagerThrottlingInterceptor()));
+    }
+
+
+    static CloudPool createIfNotExistPaaSPool(String poolId) throws Exception {
+        // Create a pool with 3 Small VMs
+        String poolVmSize = "Small";
+        int poolVmCount = 3;
+        String poolOsFamily = "4";
+        String poolOsVersion = "*";
+
+        // 10 minutes
+        long poolSteadyTimeoutInSeconds = 10 * 60 * 1000;
+
+        // Check if pool exists
+        if (!batchClient.poolOperations().existsPool(poolId)) {
+            // Use PaaS VM with Windows
+            CloudServiceConfiguration configuration = new CloudServiceConfiguration();
+            configuration.withOsFamily(poolOsFamily).withOsVersion(poolOsVersion);
+
+            List<UserAccount> userList = new ArrayList<>();
+            userList.add(new UserAccount().withName("test-user").withPassword("kt#_gahr!@aGERDXA")
+                    .withElevationLevel(ElevationLevel.ADMIN));
+            PoolAddParameter addParameter = new PoolAddParameter().withId(poolId)
+                    .withTargetDedicatedNodes(poolVmCount).withVmSize(poolVmSize)
+                    .withCloudServiceConfiguration(configuration).withUserAccounts(userList);
+            batchClient.poolOperations().createPool(addParameter);
+        }
+
+        long startTime = System.currentTimeMillis();
+        long elapsedTime = 0L;
+        boolean steady = false;
+        CloudPool pool;
+
+        // Wait for the VM to be allocated
+        while (elapsedTime < poolSteadyTimeoutInSeconds) {
+            pool = batchClient.poolOperations().getPool(poolId);
+            if (pool.allocationState() == AllocationState.STEADY) {
+                steady = true;
+                break;
+            }
+            System.out.println("wait 30 seconds for pool steady...");
+            Thread.sleep(30 * 1000);
```

Comment:
nit: replace 1000 with constant ( multiple locations )

Reply:
Refactored it into the constant SEC_TO_MILLIS

Reply:
Personally I'm not a fan of this - I would rather just have 1000 constant. I think people should be familiar with the fact that Thread.sleep operates in milliseconds, and this constant to me doesn't add any value.